### DISCUSSION TITLE
Add autofinding Proton directory in Heroic for Ubuntu and derivatives

### DISCUSSION THREAD

**Discussion by zvbarev** [Heroic-Games-Launcher/HeroicGamesLauncher#222] [Q&A]
This bug is related to https://github.com/flavioislima/HeroicGamesLauncher/issues/188

Currently Heroic doesn't support Proton install directory in Ubuntu. This is bad because Ubuntu and its derivatives market share is about 80% approximately in Linux.

My idea is finding Steam Proton automatically on the very first start of Heroic. Something like window "Wait, finding Proton...". During that process Heroic makes search of Proton in array of predefined directories for main Linux distros. If it could not find user see the message "Can't find Proton. You can add it manually by a button "Add Proton folder"" 
After that saving settings. Voila)

**Comment by flavioislima** [comment]
Hmm, but Heroic is already doing that.
I could not manage to find a way of getting where Proton is installed in case it is on another partition than the root one.
But Heroic is searching for Proton on it's default path, that is `.local/share/Steam` like I said on #188.

In the future, probably soon, Heroic will have an option to choose where Proton is installed and then this will make everyone lives easier.

For now you could just do:
`ln -s {path to proton folder} .config/heroic/tools/proton/`

Then, going to the wine settings you should see the proton version.

With an example. In case you Proton 5.13 is installed under /mnt/HDD...:
`ln -s /mnt/HDD/MyGames/Steam/common/steamapps/Proton 5.13 .config/heroic/tools/proton/`

The you should see Proton 5.13 on wine settings.

**Comment by zvbarev** [comment]
@flavioislima as I previously told I installed Steam in Ubuntu via Software Centre with settings by default.
My proton folder is `.steam/steam/steamapps/common` but Heroic doesn't use this folder.

It is DEFAULT Ubuntu path. It is on the same partition, not another. And Heroic doesn't try this path.

Thank you for workaround with ln

**Comment by ianwijma** [comment]
I used POP_OS, and my steam was also installed in `.steam/steam/steamapps/common` and their compatibility tools are in `.steam/root/compatibilitytools.d` 

Edit: For me it does find my protons in both folders. So you might want to check the latest release.

**Comment by flavioislima** [comment]
@Sebastian1978 can you try to install steam from the steam website instead of the one on the ubuntu software center?

I was reading somethings about steam on ubuntu and saw some people complaining that the version on the software center is outdated and have bugs. They recommended to install from the steam website that you always have the latest and optimized version.

I talked with some ubuntu users and they are using heroic and proton appears for them without issue. So, when you have time, try that please.
